packages/compass-e2e-tests/helpers/smoke-test/downloads.ts

```diff
@@ -0,0 +1,58 @@
+import assert from 'node:assert';
+import fs from 'node:fs';
+import path from 'node:path';
+import stream from 'node:stream';
+import { type fetch as undiciFetch } from 'undici-types';
+
+// Hacking types here because the DOM types doesn't match the fetch implemented by Node.js
+const fetch = globalThis.fetch as unknown as typeof undiciFetch;
```

Comment:
This could be avoided if we moved the code into its own package with types properly locked down.